util/topsql/stmtstats/stmtstats.go

```diff
@@ -16,30 +16,54 @@ package stmtstats
 
 import (
 	"sync"
+	"time"
 
 	"go.uber.org/atomic"
 )
 
 var _ StatementObserver = &StatementStats{}
 
+// nowFunc returns the current time; it's overridden in tests.
+var nowFunc = time.Now
+
 // StatementObserver is an abstract interface as a callback to the corresponding
 // position of TiDB's SQL statement execution process. StatementStats implements
 // StatementObserver and performs counting such as SQLExecCount/SQLDuration internally.
 // The caller only needs to be responsible for calling different methods at the
 // corresponding locations, without paying attention to implementation details.
 type StatementObserver interface {
-	// OnExecutionBegin should be called before statement execution.
-	OnExecutionBegin(sqlDigest, planDigest []byte)
+	// OnParseBegin should be called on statement begin to parse.
```

Comment:
OnRecieveCmd()

Reply:
state: initial -> cmd recieve (reset state)  -> sql/plan digest raedy -> exec begin -> finish     , if state not right, log it out.